sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -1384,4 +1384,14 @@ class FilterPushdownSuite extends PlanTest {
       condition = Some("x.a".attr === "z.a".attr)).analyze
     comparePlans(optimized, correctAnswer)
   }
+
+  test("SPARK-28220: Push down the foldable predicate to both sides of Join") {
+    val x = testRelation.subquery('x)
+    val y = testRelation.subquery('y)
+    val originalQuery = x.join(y, condition = Some(false))
+
+    val optimized = Optimize.execute(originalQuery.analyze)
+    val correctAnswer = x.where(false).join(y.where(false), condition = None).analyze
+    comparePlans(optimized, correctAnswer.analyze)
```

Comment:
We use `Inner` join to do test because it need `PruneFilters` if use `LeftOuter` join:
```scala
val x = testRelation.subquery('x)
val y = testRelation.subquery('y)
val originalQuery = x.join(y, LeftOuter, condition = Some(false))

val optimized = Optimize.execute(originalQuery.analyze)
val correctAnswer = x.join(y.where(false), LeftOuter, condition = None).analyze
comparePlans(optimized, correctAnswer.analyze)
```
Error:
```
== FAIL: Plans do not match ===
!'Join LeftOuter, false                            'Join LeftOuter
 :- LocalRelation <empty>, [a#0, b#0, c#0]         :- LocalRelation <empty>, [a#0, b#0, c#0]
 +- Filter false                                   +- Filter false
!   +- Filter false                                   +- LocalRelation <empty>, [a#0, b#0, c#0]
!      +- LocalRelation <empty>, [a#0, b#0, c#0]   
```